### PR TITLE
gpu_capture: pass gctx to gpu_capture_create()

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

**Comment by jmoguillansky-gpsw** [comment]
it might be better to keep the capture code more decoupled from the node.gl code, so it's more generic and reusable, and modular.
The capture code doesn't need the whole gctx.  It just needs a device handle

**Review by jmoguillansky-gpsw — DISMISSED** [review]

**Review by jmoguillansky-gpsw — COMMENTED** [review]
is there a way to tell if a gctx is a gl context, vulkan context, etc (e.g. if a gctx is a gctx_vk, etc)?

**Review by jmoguillansky-gpsw — DISMISSED** [review]

File: libnodegl/backends/vk/gctx_vk.c [hunk 1]
```diff
@@ -545,7 +545,7 @@ static int vk_init(struct gctx *s)
     const char *var = getenv("NGL_GPU_CAPTURE");
     s->gpu_capture = var && !strcmp(var, "yes");
     if (s->gpu_capture) {
-        s->gpu_capture_ctx = gpu_capture_ctx_create();
+        s->gpu_capture_ctx = gpu_capture_ctx_create(&s);
```
Comment (jmoguillansky-gpsw) [inline libnodegl/backends/vk/gctx_vk.c:548]: this should be s, not &s.

File: libnodegl/backends/gl/gctx_gl.c [hunk 2]
```diff
@@ -394,7 +394,7 @@ static int gl_init(struct gctx *s)
     const char *var = getenv("NGL_GPU_CAPTURE");
     s->gpu_capture = var && !strcmp(var, "yes");
     if (s->gpu_capture) {
-        s->gpu_capture_ctx = gpu_capture_ctx_create();
+        s->gpu_capture_ctx = gpu_capture_ctx_create(&s);
```
Comment (jmoguillansky-gpsw) [inline libnodegl/backends/gl/gctx_gl.c:397]: this should be s, not &s

File: libnodegl/backends/vk/gctx_vk.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/vk/gctx_vk.c]: fixed, thanks.

File: libnodegl/backends/gl/gctx_gl.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/gctx_gl.c]: fixed.

**Pull request merged**